app/scripts/controllers/createFromURL.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateFromURLController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateFromURLController', function ($scope, $routeParams, $location, $filter, AuthService, DataService, AlertMessageService, Navigate, ProjectsService ) {
+    AuthService.withUser();
+
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+
+    $scope.alerts = {};
+    $scope.selected = {};
+
+    var alertInaccessibleNamespace = function(namespace, e) {
+      $scope.alerts.inaccessibleNamespace = {
+        type: "error",
+        message: "You do not have access to the namespace \"" + namespace + "\".",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidImageStream = function(imageStream, e) {
+      $scope.alerts.invalidImageStream = {
+        type: "error",
+        message: "The requested image stream \"" + imageStream + "\" is not valid.",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidImageTag = function(imageTag, e) {
+      $scope.alerts.invalidImageTag = {
+        type: "error",
+        message: "The requested image stream tag \"" + imageTag + "\" is not valid.",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidNamespace = function(namespace) {
+      $scope.alerts.invalidNamespace = {
+        type: "error",
+        message: namespace ?
+                  "Resources from the namespace \"" + namespace + "\" are not permitted." :
+                  "A valid namespace is required."
+      };
+    };
+
+    var alertInvalidTemplate = function(template, e) {
+      $scope.alerts.invalidTemplate = {
+        type: "error",
+        message: "The requested template \"" + template + "\" is not valid.",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertResourceRequired = function() {
+      $scope.alerts.resourceRequired = {
+        type: "error",
+        message: "A valid image stream or template is required."
+      };
+    };
+
+    var showInvalidResource = function() {
+      $scope.alerts.invalidResource = {
+        type: "error",
+        message: "Image streams and templates cannot be combined."
+      };
+    };
+
+    var getTemplateParamsMap = function () {
+      try {
+        return JSON.parse($routeParams.templateParamsMap);
+      }
+      catch (e) {
+        $scope.alerts.invalidTemplateParams = {
+          type: "error",
+          message: "The template parameters are not valid."
```

Comment:
We should say specifically that it's not valid JSON and ideally show the syntax error details from `e` if present.